### PR TITLE
ReportIssue: Replace Diagnostic.Create with secondary locations

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Follow up of https://github.com/SonarSource/sonar-dotnet/pull/9298 to prevent AD0001s like in #9323 

This PR replaces usage of `Diagnostic.Create` overloads that
* Did not use the safe `DiagnosticDescriptorExtensions.CreateDiagnostic` (replaced in #9346)
* Used secondary locations parameter

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/AspNet/RouteTemplateShouldNotStartWithSlashBase.cs [hunk 1]
```diff
@@ -75,12 +75,10 @@ private void ReportIssues(SonarSymbolReportingContext context, INamedTypeSymbol
             ? MessageOnlyActions
             : MessageActionsAndController;
 
-        var secondaryLocations = actions.SelectMany(x => x.RouteParameters.Keys);
-        foreach (var classDeclaration in controllerSymbol.DeclaringSyntaxReferences.Select(x => x.GetSyntax()))
+        var secondaryLocations = actions.SelectMany(x => x.RouteParameters.Keys).Select(x => x.ToSecondary());
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/AspNet/RouteTemplateShouldNotStartWithSlashBase.cs:78]: This is about the 10th time in this PR that I see a `Select( x=> x.ToSecondary())`. Consider adding `ToSecondary()` extension methods for `IEnumerable<SyntaxNode|SyntaxToken|Location>` as well.

File: analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs [hunk 2]
```diff
@@ -38,14 +38,11 @@ public abstract class UseWhereBeforeOrderByBase<TSyntaxKind, TInvocation> : Sona
                 && Language.Syntax.TryGetOperands(invocation, out var left, out var right)
                 && LeftHasCorrectName(left, out var orderByMethodDescription)
                 && MethodIsLinqExtension(left, c.SemanticModel)
-                && MethodIsLinqExtension(right, c.SemanticModel))
+                && MethodIsLinqExtension(right, c.SemanticModel)
+                && Language.Syntax.NodeIdentifier(right) is { } rightIdentifier
+                && Language.Syntax.NodeIdentifier(left) is { } leftIdentifier)
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs:43]: This is a change in logic. It suppresses issues where the leftIdentifier is null. Before, it was raised with an invalid/ignored secondary location, and now it is filtered out.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
There is a change in logic that we should avoid.

**Comment by pavel-mikula-sonarsource** [comment]
Build doesn't work due to issue with VM. Master is also broken

File: analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs (hunk 2 above)
Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs:43]: Okay. But this change needs peach validation, then.

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs:43]: Yes, good point!

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**